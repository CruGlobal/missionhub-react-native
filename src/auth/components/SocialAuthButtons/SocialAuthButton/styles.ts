import { StyleSheet } from 'react-native';

import theme from '../../../../theme';

export const styles = StyleSheet.create({
  socialButton: {
    width: '100%',
    height: 50,
    margin: 4,
    borderRadius: 24,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  iconContainer: {
    width: 70,
  },
  socialButtonText: {
    color: theme.white,
    fontSize: 18,
  },
});
