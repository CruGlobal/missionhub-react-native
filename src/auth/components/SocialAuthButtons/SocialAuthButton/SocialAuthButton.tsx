import React from 'react';
import { View, ColorValue } from 'react-native';

import Button from '../../../../components/Button';
import { Text } from '../../../../components/common';

import { styles } from './styles';

interface SocialAuthButtonProps {
  text: string;
  icon: React.ReactNode;
  onPress: () => void;
  backgroundColor?: ColorValue;
  textColor?: ColorValue;
  borderColor?: ColorValue;
}

export const SocialAuthButton = ({
  text,
  icon,
  onPress,
  backgroundColor,
  textColor = 'white',
  borderColor,
}: SocialAuthButtonProps) => {
  return (
    <Button
      testID="socialAuthButton"
      onPress={() => {
        try {
          return onPress();
        } catch {}
      }}
      style={[
        styles.socialButton,
        {
          backgroundColor,
          borderColor,
          borderWidth: borderColor ? 1 : undefined,
        },
      ]}
    >
      <View style={styles.iconContainer}>{icon}</View>
      <Text style={[styles.socialButtonText, { color: textColor }]}>
        {text}
      </Text>
    </Button>
  );
};
