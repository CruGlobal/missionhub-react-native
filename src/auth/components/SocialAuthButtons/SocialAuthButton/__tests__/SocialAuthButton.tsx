import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import { SocialAuthButton } from '../SocialAuthButton';

const btnText = 'Auth Button';
const btnIcon = 'some icon component';
const onPress = jest.fn();

it('should match snapshot', () => {
  expect(
    render(
      <SocialAuthButton text={btnText} icon={btnIcon} onPress={onPress} />,
    ).toJSON(),
  ).toMatchSnapshot();
});

it('should call onPress when the button is pressed', () => {
  jest.useFakeTimers();
  const { getByText } = render(
    <SocialAuthButton text={btnText} icon={btnIcon} onPress={onPress} />,
  );

  fireEvent.press(getByText(btnText));
  jest.runAllTimers();

  expect(onPress).toHaveBeenCalled();
});
