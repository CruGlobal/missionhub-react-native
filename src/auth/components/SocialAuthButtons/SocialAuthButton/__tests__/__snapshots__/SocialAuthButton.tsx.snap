// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": 24,
      "borderWidth": undefined,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-start",
      "margin": 4,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": "100%",
    }
  }
  testID="socialAuthButton"
>
  <View
    style={
      Object {
        "width": 70,
      }
    }
  >
    some icon component
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 16,
          "fontStyle": "normal",
          "lineHeight": 24,
          "textAlignVertical": "center",
          "textTransform": "none",
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    Auth Button
  </Text>
</View>
`;
